src/targets/gpu/mlir.cpp

```diff
@@ -168,10 +180,19 @@ struct mlir_program
           location(mlirLocationUnknownGet(ctx.get())),
           mmodule(mlirModuleCreateEmpty(location))
     {
+#ifdef MIGRAPHX_MLIR_REGISTER_ROCMLIR
+        MlirDialectRegistry registry = mlirDialectRegistryCreate();
+        mlirRegisterRocMLIRDialects(registry);
+        mlirRegisterAllDialects(registry);
+        mlirContextAppendDialectRegistry(ctx.get(), registry);
+        mlirContextLoadAllAvailableDialects(ctx.get());
+        mlirDialectRegistryDestroy(registry);
+        mlirContextSetAllowUnregisteredDialects(ctx.get(), true /*allow*/);
+#else
```

Comment:
Why do we need the ifdef? We aren't supporting different MLIR versions.

Reply:
You're right, I'll just check the version and undefine `MIGRAPHX_MLIR` with warning message.

Reply:
I figured we should have the ifdef to match MIOpen's practice of being able to build with slightly older librockcompiler

Reply:
This code directly integrates librockcompiler, not using MIOpen. Officially, corresponding release version is the only correct version to use.